load_tests/load_tests/socketio_locustfile.py

```diff
@@ -0,0 +1,59 @@
+import os
+import time
+from pathlib import Path
+from typing import List
+
+from locust import Locust, TaskSet, between, task
+from rasa_integration_testing.helper import generate_tracker_id_from_scenario_name
+from rasa_integration_testing.interaction import InteractionLoader
+from rasa_integration_testing.scenario import (
+    Scenario,
+    ScenarioFragmentLoader,
+    load_scenarios,
+)
+from socketio import Client
+
+from load_tests.constants import SESSION_ID_KEY
+from load_tests.runner import LocustRunner
+
+ENV_INTEGRATION_TEST_PATH = "INTEGRATION_TEST_PATH"
+ENV_LOAD_TEST_TARGET_URL = "LOAD_TEST_TARGET_URL"
+
+SCENARIOS_GLOB = "*.yml"
+SCENARIOS_FOLDER = "scenarios"
+TEST_CONFIG_FILE = "config.ini"
+
+tests_path = os.environ.get(ENV_INTEGRATION_TEST_PATH, "../integration-tests-en")
+folder_path = Path(tests_path)
+scenarios_path = folder_path.joinpath(SCENARIOS_FOLDER)
+scenarios: List[Scenario] = load_scenarios(scenarios_path, SCENARIOS_GLOB)
+
+runner: LocustRunner = LocustRunner(
+    InteractionLoader(folder_path), ScenarioFragmentLoader(folder_path),
+)
+
+
+def _create_scenario_task(scenario: Scenario):
+    @task
+    def _scenario_task(self) -> None:
+        session_id = generate_tracker_id_from_scenario_name(time.time(), scenario.name)
+        self.socketio.emit("session_request", {SESSION_ID_KEY: session_id})
+        runner.run(scenario, self.socketio, session_id)
+
+    return _scenario_task
+
+
+class IntegrationTestingSet(TaskSet):
+    def on_start(self):
+        self.socketio = Client()
+        self.socketio.connect(os.environ.get(ENV_LOAD_TEST_TARGET_URL))
+
+    def on_quit(self):
+        self.socketio.disconnect()
+
+    tasks = [_create_scenario_task(scenario) for scenario in scenarios]
+
+
+class User(Locust):
+    task_set = IntegrationTestingSet
+    wait_time = between(0, 5)
```

Comment:
I guess ramp-up parameters should be configurable?

Reply:
Added optional env variables for these with examples in `.env.example`, they default to these values still.